src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -1107,4 +1111,169 @@ public LeafRope flatten(Rope rope) {
 
     }
 
+    @NodeChildren({
+            @NodeChild(type = RubyNode.class, value = "a"),
+            @NodeChild(type = RubyNode.class, value = "b")
+    })
+    public abstract static class EqualNode extends RubyNode {
```

Comment:
`Rope#equals` does not compare the encoding because it breaks the symbol table. This is something we inherited from `ByteList` that we've just never fixed:

https://github.com/graalvm/truffleruby/blob/master/src/main/java/org/truffleruby/core/rope/Rope.java#L124-L125

Having said that, in cases not involving the symbol table, we should take the encoding into account. The functionality of the guard and `Rope#equals` cannot be combined without that underlying bug being fixed. So either it should be fixed now, the guard should be reinstated, or two equality nodes should exist.